app/src/main/java/org/kiwix/kiwixmobile/wifi_hotspot/WifiHotspotManager.java

```diff
@@ -0,0 +1,88 @@
+package org.kiwix.kiwixmobile.wifi_hotspot;
+
+import android.net.wifi.WifiConfiguration;
+import android.net.wifi.WifiManager;
+import android.os.Build;
+import android.os.Handler;
+import android.util.Log;
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+import javax.inject.Inject;
+
+/**
+ * WifiHotstopManager class makes use of the Android's WifiManager and WifiConfiguration class
+ * to implement the wifi hotspot feature.
+ * Created by Adeel Zafar on 28/5/2019.
+ */
+
+public class WifiHotspotManager {
+  private static final String TAG = "WifiHotspotManager";
+  private WifiManager.LocalOnlyHotspotReservation hotspotReservation;
+  private HotspotStateListener hotspotStateListener;
+
+  @Inject
```

Comment:
this should not be annotated `@Inject`, you annotate fields with `@Inject` when you want to do `field injection` ie an explicit call to `SomeComponent.inject(this)`.
You are performing `constructor injection` seeing as your constructor is annotated `@Inject`

Reply:
Got it. Fixed.